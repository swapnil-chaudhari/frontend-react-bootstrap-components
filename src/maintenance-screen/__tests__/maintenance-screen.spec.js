import React from 'react';
import renderShallow from 'render-shallow';
import MaintenanceScreen from '../maintenance-screen';

describe('<MaintenanceScreen>', () => {
    describe('when rendered with default props', () => {
        let component;
        const props = {
            title: {
                text: 'We\'re down for maintenance. We\'ll be back up by',
                time: '2 AM ',
                timeZone: 'EST.'
            },
            subTitle: 'Sorry for any inconvenience.'
        };

        beforeAll(() => {
            component = renderShallow(<MaintenanceScreen { ...props } />).output;
        });

        test('contains logo', () => {
            expect(component).toMatchSnapshot();
        });

        test('contains Maintenance icon and text', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
