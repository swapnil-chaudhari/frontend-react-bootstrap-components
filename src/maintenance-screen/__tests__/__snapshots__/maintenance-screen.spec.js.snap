// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaintenanceScreen> when rendered with default props contains Maintenance icon and text 1`] = `
<div
  className="MaintenanceScreen"
>
  <div
    className="MaintenanceScreen-logo"
  >
    <LogoAndName />
  </div>
  <div
    className="MaintenanceScreen-content"
  >
    <div
      className="MaintenanceScreen-graphic"
    >
      <Maintenance />
    </div>
    <p
      className="MaintenanceScreen-title"
    >
      We're down for maintenance. We'll be back up by
       
      <span
        className="MaintenanceScreen-highlighted"
      >
        2 AM 
      </span>
      EST.
    </p>
    <p
      className="MaintenanceScreen-subtitle"
    >
      Sorry for any inconvenience.
    </p>
  </div>
</div>
`;

exports[`<MaintenanceScreen> when rendered with default props contains logo 1`] = `
<div
  className="MaintenanceScreen"
>
  <div
    className="MaintenanceScreen-logo"
  >
    <LogoAndName />
  </div>
  <div
    className="MaintenanceScreen-content"
  >
    <div
      className="MaintenanceScreen-graphic"
    >
      <Maintenance />
    </div>
    <p
      className="MaintenanceScreen-title"
    >
      We're down for maintenance. We'll be back up by
       
      <span
        className="MaintenanceScreen-highlighted"
      >
        2 AM 
      </span>
      EST.
    </p>
    <p
      className="MaintenanceScreen-subtitle"
    >
      Sorry for any inconvenience.
    </p>
  </div>
</div>
`;
